src/UnifiedFFI/ExternalObject.extension.st

```diff
@@ -2,8 +2,9 @@ Extension { #name : #ExternalObject }
 
 { #category : #'*UnifiedFFI' }
 ExternalObject >> autoRelease [
-	self isExternalAddress ifFalse: [ ^ self ].
-	^ self getHandle autoRelease
+	self getHandle isExternalAddress ifFalse: [ ^ self ].
+	self getHandle autoRelease.
+	^ self
```

Comment:
this fix is the only change that can affect users that don't explicitly set the new option